### ISSUE TITLE
[BUILD] Error during compilation

### ISSUE THREAD

**Issue by jack9603301** [AcademySoftwareFoundation/OpenImageIO#4146]
When I compile using command like below:

```
cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_LIBDIR=/usr/lib -DDCMTK_FORCE_FPIC_ON_UNIX=ON -DBUILD_TESTING=OFF
```

build errors: 

```
/usr/bin/ld: ../../lib/libOpenImageIO.so.2.5.7: undefined reference to `OFiconv_open'
/usr/bin/ld: ../../lib/libOpenImageIO.so.2.5.7: undefined reference to `set_oficonv_logger_callback'
/usr/bin/ld: ../../lib/libOpenImageIO.so.2.5.7: undefined reference to `OFiconv_close'
/usr/bin/ld: ../../lib/libOpenImageIO.so.2.5.7: undefined reference to `OFiconv'
/usr/bin/ld: ../../lib/libOpenImageIO.so.2.5.7: undefined reference to `get_oficonv_logger_callback'
/usr/bin/ld: ../../lib/libOpenImageIO.so.2.5.7: undefined reference to `OFlocale_charset'
/usr/bin/ld: ../../lib/libOpenImageIO.so.2.5.7: undefined reference to `OFiconvctl'
```

**Comment by lgritz** [comment]
I suspect that you are finding a static version of dcmtk, which is somehow not being linked properly?

I'm a little suspicious of `-DDCMTK_FORCE_FPIC_ON_UNIX=ON`, which is not a cmake variable name that OpenImageIO uses. So I'm wondering where that came from and why you added it. 

Where you say "when I compile using...", is that the command you are using to compile the OpenImageIO project itself? Or DCMTK? Or something else?

Can you attach a full verbose build log?

**Comment by lgritz** [comment]
I might have a fix? Please try #4147, if you're able to apply that patch, and see if it helps?

**Comment by jack9603301** [comment]
Please ignore it because that variable is for DCMTK and I didn't delete it.

**Comment by jack9603301** [comment]
> I might have a fix? Please try #4147, if you're able to apply that patch, and see if it helps?

I recompiled dcmtk as a dynamic library and then applied your patch. The compilation is normal now.
